bin/genesis

```diff
@@ -4249,12 +4454,21 @@ sub {
 	if (-f $new_script) {
 		chmod_or_fail(0755, $new_script) unless -x $new_script;
 		my $curdir = Cwd::getcwd;
-		my $out = qx(cd $kit_dir && $new_script $curdir $name);
-		die "Could not create new environment $name using kit $kit/$version"
-			if ($! >> 8);
+		my $helper = new_script_helper($kit,$version);
+		my $script = clean_heredoc(<<"			|EOF");
+			|cd $kit_dir
+			|source "$helper"
+			|hooks/new "$curdir" "$name" "$prefix"
+			|EOF
+
+		system($script);
+		if ($? >> 8) {
+			error "#R{ERROR}: Cound not create new environment $name by executing $kit/$version hooks/new script";
+			exit 2;
+		}
 		die "New environment '$name' was not generated.  Contact your kit developer for a fix"
 			unless -f "$name.yml";
-		@features = get_key($name, 'kit.features', []);
+		@features = @{get_key($name, 'kit.features', [])};
```

Comment:
This sounds good in theory, but in practice, the commonality of the following makes this much harder:
`get_key($file, "kit.features", get_key($file, "kit.subkits", []))` 

...because the chained call triggers wantarray, which means the "default" scalar  gets expanded

Reply:
Fair enough.